rpcs3/Emu/Cell/Modules/cellAudio.cpp

```diff
@@ -1264,13 +1280,20 @@ error_code cellAudioGetPortTimestamp(u32 portNum, u64 tag, vm::ptr<u64> stamp)
 
 	if (port.global_counter < tag)
 	{
-		cellAudio.error("cellAudioGetPortTimestamp(portNum=%d, tag=0x%llx, stamp=*0x%x) -> CELL_AUDIO_ERROR_TAG_NOT_FOUND", portNum, tag, stamp);
 		return CELL_AUDIO_ERROR_TAG_NOT_FOUND;
 	}
 
-	u64 delta_tag = port.global_counter - tag;
-	u64 delta_tag_stamp = delta_tag * g_audio->cfg.audio_block_period;
-	*stamp = port.timestamp - delta_tag_stamp;
+	if (stamp)
```

Comment:
honestly its better to access violate as its more alarming to user than a warning, especially if thats exactly what happens on real ps3. in other words if theres no error check nor documentation on it it might not check at all on real ps3 and access violate.
Continuing after nullptr dereference is a bad hack in general.

Reply:
will a game crash if this happens, or can it recover/catch ?

Reply:
if the firmware doesnt check it it will crash. Unlike lv2 which uses "safe" memory access methods, modules access memory the same way as games' code which means they crash.
This is why lv2 has EFAULT error code and not just null argument error as it can detect all kinds of invalid addresses.
Modules cannot recover from invalid memory access.